src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala

```diff
@@ -295,6 +303,21 @@ trait ClusterComponent extends LeoComponent {
         .result map { recs => unmarshalFullCluster(recs)}
     }
 
+    // TODO: find how to make these not full cluster queries
+    def getClustersWithWelderEnabledByProject(project: GoogleProject): DBIO[Seq[Cluster]] = {
+      fullClusterQuery
+        .filter { _._1.googleProject === project.value }
+        .filter { _._1.welderEnabled === true }
+        .result map { recs => unmarshalFullCluster(recs)}
```

Comment:
I think you only need to know whether there exists cluster. Something like `def existClustersWithWelderEnabled(project: GoogleProject): DBIO[Boolean]`

Reply:
I'm curious why this requirement exists in the first place..

Reply:
There has been some face-talking about this so I don't know where it landed but IMO this is added complexity for no benefit. AoU is going to recreate all clusters to use welder and from that point on they will only be creating welder enabled clusters. Once this is enabled in Terra, we will only be creating welderEnabled clusters across the entire platform and the flag will be taken out. AoU and Terra projects do not mix so there should be no scenario where this is a legitimate concern.

If there is a benefit and I am missing it (or if my assumptions above no longer hold true), please fill me in.

There was a discussion about it in the leo-data-syncing channel a few weeks ago and we never came to an agreement. See https://bvdp.slack.com/archives/CJMDJUJ06/p1560169892067400